src/nrfx/hal/nrf_clock.c

```diff
@@ -32,14 +32,62 @@ void nrf_clock_task_trigger(NRF_CLOCK_Type * p_reg, nrf_clock_task_t task)
 {
   if (task == NRF_CLOCK_TASK_HFCLKSTART) {
     NRF_CLOCK_regs.TASKS_HFCLKSTART = 1;
+    NRF_CLOCK_regs.HFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTART();
   } else if (task == NRF_CLOCK_TASK_HFCLKSTOP) {
     NRF_CLOCK_regs.TASKS_HFCLKSTOP = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTOP();
   } else if (task == NRF_CLOCK_TASK_LFCLKSTART) {
     NRF_CLOCK_regs.TASKS_LFCLKSTART = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_LFCLKSTART();
+  } else if (task == NRF_CLOCK_TASK_LFCLKSTOP) {
+    NRF_CLOCK_regs.TASKS_LFCLKSTOP = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 0;
```

Comment:
Following what is described above, you would need to create the sideeffect function in NRF_clock.c and clear that LFCLKRUN there.
In any case, note that the models did not expect anywhere the LF clock would ever be turned off. Meaning for example the RTC model assumes its LFCLK input keeps running forever.
What is the use case in which this needs to be modeled?

Reply:
I'd ask you to create a new `nrf_clock_reqw_sideeffects_TASK_LFCLKSTOP` next to nrf_clock_reqw_sideeffects_TASKS_LFCLKSTART() and set that register to 0 there, with a comment in the code describing that no effect of turning the clock off is actually modeled